src/Analysis/Ast/Impl/Analyzer/Handlers/AssignmentHandler.cs

```diff
@@ -71,12 +71,26 @@ internal sealed class AssignmentHandler : StatementHandler {
                 }
 
                 var source = value.IsGeneric() ? VariableSource.Generic : VariableSource.Declaration;
-                Eval.DeclareVariable(ne.Name, value ?? Module.Interpreter.UnknownType, source, Eval.GetLocationOfName(ne));
+                var location = Eval.GetLocationOfName(ne);
+                if (IsValidAssignment(ne.Name, location)) {
+                    Eval.DeclareVariable(ne.Name, value ?? Module.Interpreter.UnknownType, source, location);
+                }
             }
 
             TryHandleClassVariable(node, value);
         }
 
+        private bool IsValidAssignment(string name, Location loc) {
+            if (Eval.GetInScope(name) is ILocatedMember m) {
+                // Class and function definition are processed first, so only override
+                // if assignment happens after declaration
+                if(loc.IndexSpan.Start < m.Location.IndexSpan.Start) {
```

Comment:
Should I worry about assignment and class/function declarations on the same line? If not, then simply checking their line numbers is enough.

Reply:
Technically we may simply avoid assignments to classes and functions completely... I.e. if `A is a class`, ignore `A= ...`

Reply:
But what if we have:
```
class A: ...


A = None
A = another_class_dec
```

It seems weird from a library creator perspective, but surely it could happen.